addons/stock_account/wizard/wizard_valuation_history.py

```diff
@@ -8,23 +8,30 @@ class WizardValuationHistory(models.TransientModel):
     _name = 'wizard.valuation.history'
     _description = 'Wizard that opens the stock valuation history table'
 
-    choose_date = fields.Boolean('Inventory at Date')
-    date = fields.Datetime('Date', default=fields.Datetime.now, required=True)
+    Compute = fields.Selection([
+        (0, 'Current Inventory'),
+        (1, 'At Specific Date')
+        ])
+    date = fields.Datetime('Inventory at Date', help="Choose a date to get inventory at that date")
 
     @api.multi
     def open_table(self):
         self.ensure_one()
-        ctx = dict(
-            self._context,
-            history_date=self.date,
-            search_default_group_by_product=True,
-            search_default_group_by_location=True)
-        return {
-            'domain': "[('date', '<=', '" + self.date + "')]",
-            'name': _('Stock Value At Date'),
-            'view_type': 'form',
-            'view_mode': 'tree',
-            'res_model': 'stock.history',
-            'type': 'ir.actions.act_window',
-            'context': ctx,
-        }
+        if self.date:
```

Comment:
Check on the renamed "Compute" field, no reason to do it on date existence. 

This way we also cover the case where the user selects "at specific date", sets a date, then changes his mind and selects "Current Inventory".